modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2556,3 +2566,63 @@ def _is_trivial_index(cls, index):
             and index.min() == 0
             and index.max() == len(index) - 1
         )
+
+
+class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
+    """
+    Proxy class for lazily retrieving categorical dtypes from arrow tables.
+
+    Parameters
+    ----------
+    table : pyarrow.Table
+        Source table.
+    name : str
+        Column name.
+    """
+
+    def __init__(self, table: pyarrow.Table, name: str):
```

Comment:
```suggestion
    def __init__(self, table: pyarrow.Table, column_name: str):
```
This may seem as a name for the table.